src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -209,6 +215,7 @@ public void createIndices(Set<String> indices) {
               throw new ConnectException("Could not create index '" + index + "'" + msg);
             }
           }
+          indexCache.put(index, true);
```

Comment:
we should check if `indexCache` contains any or all of indices before executing operations on the ES cluster.

Reply:
👍